### ISSUE TITLE
Handle multiple invocation of credential plugin

### ISSUE THREAD

**Issue by int128** [int128/kubelogin#389]
## Purpose of the feature (why)
When we run kubectl commands in parallel, each process runs the credential plugin. It may cause the unavailable port error. For example, when I run 3 kubectl commands in parallel:

```console
% kubectl get nodes

% kubectl get nodes

% kubectl get nodes
error: get-token: could not get a token: authentication error: authcode-browser error: authentication error: authorization code flow error: oauth2 error: authorization error: could not start a local server: no available port: could not listen: listen tcp 127.0.0.1:8000: bind: address already in use, could not listen: listen tcp 127.0.0.1:18000: bind: address already in use
Unable to connect to the server: getting credentials: exec: executable kubectl failed with exit code 1
```

It would be nice if other processes wait for the first authentication.

## Your idea (how)
When the port is already used, try reading the token cache until a timeout.

**Comment by int128** [comment]
- Method 1: Create a lockfile, wait for the lockfile if exists
- Method 2: Wait for the token cache if the local server port is in use

**Comment by eric-poitras** [comment]
Hi sir! 

I gave it a shot at implementing this feature. I used method 1 since it seemed more simple and straightforward. I am quite new to GO so all review and feedback is greatly apreciated! 

Thanks and have a nice day !

**Comment by int128** [comment]
Available in v1.22.0.
